packages/framer-motion/src/animation/optimized-appear/start.ts

```diff
@@ -2,21 +2,60 @@ import { appearStoreId } from "./store-id"
 import { animateStyle } from "../waapi"
 import { NativeAnimationOptions } from "../waapi/types"
 import { optimizedAppearDataId } from "./data-id"
+import { handoffOptimizedAppearAnimation } from "./handoff"
+import { appearAnimationStore } from "./store"
 
 export function startOptimizedAppearAnimation(
     element: HTMLElement,
     name: string,
     keyframes: string[] | number[],
-    options: NativeAnimationOptions
-): Animation | undefined {
-    window.MotionAppearAnimations ||= new Map()
-
+    options: NativeAnimationOptions,
+    onReady?: (animation: Animation) => void
+): void {
     const id = element.dataset[optimizedAppearDataId]
-    const animation = animateStyle(element, name, keyframes, options)
 
-    if (id && animation) {
-        window.MotionAppearAnimations.set(appearStoreId(id, name), animation)
+    if (!id) return
+
+    window.HandoffAppearAnimations = handoffOptimizedAppearAnimation
+
+    const storeId = appearStoreId(id, name)
+
+    /**
+     * Use a dummy animation to detect when Chrome is ready to start
+     * painting the page and hold off from triggering the real animation
+     * until then.
+     */
+    const readyAnimation = animateStyle(
+        element,
+        name,
+        [keyframes[0] as number, keyframes[0] as number],
+        { duration: 10000, ease: "linear" }
```

Comment:
Why is the duration 10 secs here?

Reply:
I'll add a comment, it's basically long enough for a `ready` to fire - obviously this will usually be low `ms`

Reply:
Ah, right, so we're waiting until the _start_ of this "fake" animation, right? Yeah, we should add a comment about that.